app/api/src/osc/osc_handler.cpp

```diff
@@ -0,0 +1,318 @@
+// Full project source: https://github.com/samaaron/sonic-pi
+// License: https://github.com/samaaron/sonic-pi/blob/main/LICENSE.md
+//
+// Copyright 2013, 2014, 2015, 2016 by Sam Aaron (http://sam.aaron.name).
+// All rights reserved.
+//
+// Permission is granted for use, copying, modification, and
+// distribution of modified versions of this work as long as this
+// notice is included.
+//++
+
+// OSC stuff
+#include <algorithm>
+#include <iostream>
+#include <cassert>
+
+#include "api/logger.h"
+#include "api/osc/osc_handler.h"
+#include "api/osc/osc_pkt.hh"
+#include "api/sonicpi_api.h"
+
+#include "api/logger.h"
+#include "api/string_utils.h"
+
+#undef max
+#undef min
+
+using namespace std::chrono;
+
+namespace SonicPi
+{
+
+OscHandler::OscHandler(IAPIClient* pClient)
+    : m_pClient(pClient)
+{
+    assert(pClient);
+}
+
+void OscHandler::oscMessage(std::vector<char> buffer)
+{
+    //QColor bg;
+
+    pr.init(&buffer[0], buffer.size());
+
+    oscpkt::Message* msg;
+    while (pr.isOk() && (msg = pr.popMessage()) != 0)
+    {
+        if (msg->match("/log/multi_message"))
+        {
+            MessageInfo message;
+            message.type = MessageType::Muti;
+
+            oscpkt::Message::ArgReader ar = msg->arg();
+            ar.popInt32(message.jobId);
+            ar.popStr(message.threadName);
+            ar.popStr(message.runtime);
+
+            int msg_count;
+            ar.popInt32(msg_count);
+
+            for (int i = 0; i < msg_count; i++)
+            {
+                MessageData messageData;
+                ar.popInt32(messageData.style);
+                ar.popStr(messageData.text);
+                message.multi.push_back(messageData);
+            }
+
+            m_pClient->Report(message);
+        }
+        else if (msg->match("/incoming/osc"))
+        {
+            std::string time;
+            int id;
+            std::string address;
+            std::string args;
+            if (msg->arg().popStr(time).popInt32(id).popStr(address).popStr(args).isOkNoMoreArgs())
+            {
+                CueInfo info;
+                info.address = address;
+                info.args = args;
+                info.id = id;
+                info.time = time;
+                info.index = m_currentQueueIndex++;
+                info.arrivalTime = std::chrono::high_resolution_clock::now();
+                m_pClient->Cue(info);
+            }
+            else
+            {
+                LOG(ERR, "Unhandled OSC msg /incoming/osc");
+            }
+        }
+        else if (msg->match("/log/info"))
+        {
+            MessageInfo message;
+            message.type = MessageType::Info;
+            if (msg->arg().popInt32(message.style).popStr(message.text).isOkNoMoreArgs())
+            {
+                LOG(DBG, "/log/info: " << message.text);
+                m_pClient->Report(message);
+            }
+            else
+            {
+                LOG(ERR, "Unhandled OSC msg /info");
```

Comment:
I realise the existing implementation on main already does this too, but it seems not ideal to have these error log messages not describing exactly the same OSC message paths that the code tries to process... eg here we try to match and process `/log/info`, but the error log will mention unhandled OSC msg `/info`.
In the final implementation of this (in this PR or later) seems to me a good idea to make these consistent.
(Same goes for a lot of the other OSC matching code elsewhere in this file).

Reply:
Sure, I can improve these messages.